### DISCUSSION TITLE
[HOW TO] MacOS: proper formatting SD card

### DISCUSSION THREAD

**Discussion by JDuinkerken** [jomjol/AI-on-the-edge-device#1401] [General] [answered]
After several tries, I got the strong impression that there was something with file I/O on the SD card. But only for the image data files, not for a file like config.ini.
- the SD cards I tried were formatted with MBR and the MS-DOS filesystem by using diskUtil under MacOS. The results were not good (in short: the system worked OK *until* the first image had to be analysed. After that, no results came and it took ages to not produce anything).
- My suspicion went to the low level file system. Did the Mac really format in MS-DOS format? Or was it some sort of look-alike filesystem?
- Long story short, eventually I used the command written below to format the SD card (from MacOS terminal command prompt) and everything is now running like a charm :-)
- The SD card was formatted with:  “newfs_msdos -F 16 (path to SD Card device)/(SD Card device number)”
(in my case: "sudo newfs_msdos -F 16 /dev/disk4")

Maybe it is of any use to you :-)

Cheers,
Johan.

**Comment by caco3** [comment, marked as answer]
Thanks for sharing your experience!
There are other reports also which indicate that the formating is essential for a proper operation :unamused:

**Comment by donnie-darko** [comment]
TL;DR never copy files to the SD Card with MacOS. 

In details:
I've spent countless hours reformatting, reinstalling, reflashing, recopying this project to an ESP-CAM bought from banggood. 
Maybe the hidden dotfiles/metadata cause something but the setup flow was painfully slow and often it never worked after the first image analyzed - the same experience that @JDuinkerken wrote. 
So why I suspect the root cause the hidden files that MacOS create on SD Card? Because regardless the OS that format the SD Card (Windows, MacOS, Linux) the experience is always the same if eventually using MacOS to copy the files to the Card.
And yes I tried to format the Card as @JDuinkerken suggested but it never helped.
So I switched to use Windows to copy the files and everything went smooth and worked since.
